debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsStreamProducerIT.java

```diff
@@ -2088,6 +2088,100 @@ public void shouldStreamEnumAsKnownType() throws Exception {
         assertThat(consumer.isEmpty()).isTrue();
     }
 
+    // @Test
+    // public void shouldStreamEnumArrayAsKnownType() throws Exception {
+    // // Specifically enable `column.propagate.source.type` here to validate later that the actual
+    // // type, length, and scale values are resolved correctly when paired with Enum types.
+    // TestHelper.execute("CREATE TABLE enum_array_table (pk SERIAL, PRIMARY KEY (pk));");
+    // startConnector(config -> config
+    // .with(PostgresConnectorConfig.INCLUDE_UNKNOWN_DATATYPES, true)
+    // .with(PostgresConnectorConfig.SNAPSHOT_MODE, SnapshotMode.NEVER)
+    // .with("column.propagate.source.type", "public.enum_array_table.value")
+    // .with(PostgresConnectorConfig.TABLE_WHITELIST, "public.enum_array_table"), false);
+
+    // waitForStreamingToStart();
+
+    // // We create the enum type after streaming started to simulate some future schema change
+
+    // TestHelper.execute("CREATE TYPE test_type AS ENUM ('V1','V2');");
+    // TestHelper.execute("ALTER TABLE enum_array_table ADD COLUMN value test_type[] NOT NULL");
```

Comment:
This test isn't working for some reason. After adding a bunch of logging, it _looks_ like my type registry is getting cleared out or something. I see the enum array type get added to the type registry, but at some point it is no longer in my type registry. Any ideas here?

Reply:
Things seem to work fine if I create the type before the table for some reason. Creating the table and then adding the type later results in:
```
org.junit.ComparisonFailure: [field name: value] expected:<'{"type" : "[ARRAY", "optional" : "false", "value" : {"name" : "io.debezium.data.Enum", "type" : "STRING", "optional" : "false", "version" : "1"}]}'> but was:<'{"type" : "[BYTES", "optional" : "false"]}'>
```

Reply:
Aaaaah. I think I know why this doesn't work when altering the table. There's a mechanism for resolving unknown types in the type registry. It looks like it's only hooked up to work with non array types. I think that's my issue.

Reply:
Yup, that did it 😀
https://github.com/debezium/debezium/pull/1454/commits/9cf96c3b796cd7f857d5f3afd0a57d5c8cb12bd8

I'll circle back and refactor the duplicate code out once I've got things working.

Reply:
Yup, that did it 😀
https://github.com/debezium/debezium/pull/1454/commits/9cf96c3b796cd7f857d5f3afd0a57d5c8cb12bd8